packages/bff/src/utilities/fileutils.ts

```diff
@@ -0,0 +1,206 @@
+  eservice: catalogApi.EService,
+  descriptor: catalogApi.EServiceDescriptor
+) {
+  return {
+    name: eservice.name,
+    description: eservice.description,
+    technology: eservice.technology,
+    mode: eservice.mode,
+    descriptor: {
+      interface: descriptor.interface && {
+        prettyName: descriptor.interface.prettyName,
+        path: descriptor.interface.path,
+      },
+      docs: descriptor.docs.map((doc) => {
+        const uniqueName = getUniqueNameByDocumentId(
+          fileDocumentRegistry,
+          doc.id
+        );
+        return {
+          prettyname: doc.prettyName,
+          path: `documents/${uniqueName}`,
+        };
+      }),
+      audience: descriptor.audience,
+      voucherLifespan: descriptor.voucherLifespan,
+      dailyCallsPerConsumer: descriptor.dailyCallsPerConsumer,
+      dailyCallsTotal: descriptor.dailyCallsTotal,
+      description: descriptor.description,
+      agreementApprovalPolicy: descriptor.agreementApprovalPolicy,
+      riskAnalysis: eservice.riskAnalysis.map((ra) => ({
+        name: ra.name,
+        riskAnalysisForm: {
+          version: ra.riskAnalysisForm.version,
+          singleAnswers: ra.riskAnalysisForm.singleAnswers.map((sa) => ({
+            key: sa.key,
+            value: sa.value,
+          })),
+          multiAnswers: ra.riskAnalysisForm.multiAnswers.map((ma) => ({
+            key: ma.key,
+            values: ma.values,
+          })),
+        },
+      })),
+    },
+  };
+}
+
+/* 
+  This function creates a zip file fetched from the S3 bucket 
+  using FileManager, the zip file containing the following files:
+  - descriptor's interface file
+  - descriptor's documents file
+  - configuration file
+
+  The zip folder structure in output is the following:
+  - zipFolderName
+      |
+      |- interface
+      |     |
+      |     |- interfaceFile.{fileExtension}
+      |
+      |- documents
+      |     |
+      |     |- documentFile1.{fileExtension}
+      |     |- documentFile2.{fileExtension}
+      |     |- ...
+      |
+      |- configuration.json
+*/
+export async function createDescriptorDocumentZipFile(
+  s3BucketName: string,
+  fileManager: FileManager,
+  logger: Logger,
+  zipFolderName: string,
+  eservice: catalogApi.EService,
+  descriptorId: string
+): Promise<Uint8Array> {
+  const descriptor = retrieveEserviceDescriptor(eservice, descriptorId);
+  verifyExportEligibility(descriptor);
+
+  const interfaceDocument = descriptor.interface;
+  if (!interfaceDocument) {
+    throw missingInterface(eservice.id, descriptorId);
+  }
+
+  const fileDocumentRegistry = buildFileDocumentRegistry(descriptor.docs);
+  const configuration = buildJsonConfig(
+    fileDocumentRegistry,
+    eservice,
+    descriptor
+  );
+
+  const zip = new AdmZip();
+
+  // Add interface file to the zip
+  const interfaceFile = await fileManager.get(
+    s3BucketName,
+    `${interfaceDocument.path}/${interfaceDocument.name}`,
```

Comment:
```suggestion
    fileManager.buildS3Key(
      interfaceDocument.path,
      interfaceDocument.id,
      interfaceDocument.name
    ),
```
We can leverage on `buildS3Key` here too, which includes also the interface id